// grunt-traceur-compiler/Gruntfile.js
'use strict';

module.exports = function(grunt) {
    var lintFiles = ['tasks/*.js', 'Gruntfile.js'],
        traceurFiles = {},
        npmTasks = [
            'grunt-contrib-copy',
            'grunt-contrib-jshint',
            'grunt-jscs-checker',
            'grunt-mocha-test'
        ];

    grunt.file.expand('test/es6/*_test.js').map(function(name) {
        traceurFiles[name.replace('es6', 'es5')] = name;
    });

    grunt.initConfig({
        traceur: {
            options: {
                experimental: true
            },
            test: {
                files: traceurFiles
            }
        },

        mochaTest: {
            options: {
                reporter: 'spec',
                require: ['lib/traceur-runtime.js']
            },
            test: {
                src: ['test/es5/*_test.js']
            }
        },

        jscs: {
            options: {
                config: '.jscs.json'
            },
            test: lintFiles
        },

        jshint: {
            options: {
                jshintrc: '.jshintrc'
            },
            test: lintFiles.concat(['test/es6/*_test.js'])
        },

        copy: {
            test: {
                src: 'node_modules/traceur/bin/traceur-runtime.js',
                dest: 'lib/traceur-runtime.js'
            }
        }
    });

    grunt.loadTasks('tasks');

    npmTasks.forEach(function(task) {
        grunt.loadNpmTasks(task);
    });

    grunt.registerTask('default', ['jshint', 'jscs', 'copy', 'traceur', 'mochaTest']);
};
